rpc/grpc/server/services/blockservice/service.go

```diff
@@ -0,0 +1,127 @@
+package blockservice
+
+import (
+	context "context"
+	"fmt"
+
+	"github.com/cometbft/cometbft/internal/rpctrace"
+	"github.com/cometbft/cometbft/libs/log"
+	cmtpubsub "github.com/cometbft/cometbft/libs/pubsub"
+	blocksvc "github.com/cometbft/cometbft/proto/tendermint/services/block/v1"
+	"github.com/cometbft/cometbft/store"
+	"github.com/cometbft/cometbft/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type blockServiceServer struct {
+	store    *store.BlockStore
+	eventBus *types.EventBus
+	logger   log.Logger
+}
+
+// New creates a new CometBFT version service server.
+func New(store *store.BlockStore, eventBus *types.EventBus, logger log.Logger) blocksvc.BlockServiceServer {
+	return &blockServiceServer{
+		store:    store,
+		eventBus: eventBus,
+		logger:   logger.With("service", "BlockService"),
+	}
+}
+
+// GetByHeight implements v1.BlockServiceServer GetByHeight method
+func (s *blockServiceServer) GetByHeight(_ context.Context, req *blocksvc.GetByHeightRequest) (*blocksvc.GetByHeightResponse, error) {
+	var height int64
+
+	// retrieve the last height in the store
+	latestHeight := s.store.Height()
```

Comment:
A check for the lowest height can also be added here:
`earliestHeight := s.store.Base()`